backend/utils/graph_db_utils_test.py

```diff
@@ -0,0 +1,52 @@
+from unittest.mock import ANY, MagicMock
+from neo4j import Driver, Session
+import pytest
+import utils.graph_db_utils
+
+def test_database_connectivity_is_healthy(mocker):
+    mock_driver = mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
+    mock_driver.verify_connectivity.return_value = None
+
+    connected = utils.graph_db_utils.test_connection()
+
+    assert connected
+    mock_driver.verify_connectivity.assert_called_once()
+    mock_driver.close.assert_called_once()
+
+
+def test_database_connectivity_is_unhealthy(mocker):
+    mock_driver = mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
```

Comment:
this mock_driver line could be extracted to its own fixture if you wanted to, something like:

```python
@pytest.fixture
def mock_driver(mocker):
  return mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
```

and then `mock_driver` would be available as an arg to each test method, I think... 😅

Reply:
Great tip @harryBedfordSL - done. 👍